### PR TITLE
LPS-105151 SF simplify?

### PR DESCRIPTION
**Pull request by csierra**
hey @stian-sigvartsen, 
please check my last changes and if you are ok with them please forward this to Brian. 

/cc @martamedio

### CODE REVIEW

File: modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectProviderRegistryImpl.java [hunk 1]
```diff
@@ -133,13 +133,17 @@ public void updated(String pid, Dictionary<String, ?> properties) {
...
+			return;
+		}
+
 		if (oldProperties != null) {
 			long oldCompanyId = GetterUtil.getLong(
 				oldProperties.get("companyId"));
 
-			if ((companyId == CompanyConstants.SYSTEM) ||
-				(oldCompanyId == CompanyConstants.SYSTEM)) {
-
```
Comment (stian-sigvartsen) [inline modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectProviderRegistryImpl.java:111]: Hi @csierra . I think removing the first condition introduces a bug when ```oldCompanyId > 0 && companyId = 0```. This results in only the Map entry with companyId=0 being rebuilt consequently the provider is not found when looking up by any other companyId. Do you agree?

Comment (csierra) [reply modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectProviderRegistryImpl.java:111]: but I moved that condition to the top, didn't I?

Comment (csierra) [reply modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectProviderRegistryImpl.java:111]: https://github.com/stian-sigvartsen/liferay-portal/pull/250/commits/b75d345b080c83ad7f554d2f92fed1b8a391af3e#diff-ada517eca42d78e140e0631664611d66R136

isn't this equivalent?

Comment (stian-sigvartsen) [reply modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectProviderRegistryImpl.java:111]: You are correct! Sorry I thought the change was the other way around. It looks fine then.

**Comment by stian-sigvartsen** [comment]
Looks great. Sent you a squashed version for rubber stamping :) Please see csierra/liferay-portal/pull/726

**Pull request closed without merging**